sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/concatenate_data/collector_matching_strategy.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "cloud_collector.hpp"
+
+#include <rclcpp/node.hpp>
+
+#include <sensor_msgs/msg/point_cloud2.hpp>
+
+#include <list>
+#include <memory>
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace autoware::pointcloud_preprocessor
+{
+
+struct MatchingParams
+{
+  std::string topic_name;
+  double cloud_timestamp;
+  double cloud_arrival_time;
+};
+
+class CollectorMatchingStrategy
+{
+public:
+  virtual ~CollectorMatchingStrategy() = default;
+
+  [[nodiscard]] virtual std::optional<std::shared_ptr<CloudCollector>> match_cloud_to_collector(
+    const std::list<std::shared_ptr<CloudCollector>> & cloud_collectors,
+    const MatchingParams & params) const = 0;
+  virtual void set_collector_info(
+    std::shared_ptr<CloudCollector> & collector, const MatchingParams & matching_params) = 0;
+
+protected:
+  CollectorStrategyType strategy_type_;
```

Comment:
This is redundant, please remove. Please make sure once more that you are implementing polymorphism properly:

The base class shall NEVER have any dependency on specific child classes/types/etc.

Reply:
Thanks!
commit https://github.com/autowarefoundation/autoware.universe/commit/fd7db92821b09e31edda9144aaf558f58e92be6d fixed the problem.